usr/iscsi/iscsi_tcp.c

```diff
@@ -280,8 +280,18 @@ static void iscsi_tcp_event_handler(int fd, int events, void *data)
 	if (conn->state == STATE_CLOSE)
 		dprintf("connection closed\n");
 
-	if (conn->state != STATE_CLOSE && events & EPOLLOUT)
-		iscsi_tx_handler(conn);
+	if (conn->state != STATE_CLOSE && events & EPOLLOUT) {
+		if (conn->tx_task && conn->state == STATE_SCSI) {
+			do {
```

Comment:
Prasad, is this change related to send as much as possible on network once we get the EPOLLOUT event. How was the approach earlier, is it that after sending one packer over network tgt was trying to register for POLL again and POLL callback was getting triggered immediately, resulting in a loop.

Reply:
Got after discussion that it's trying to avoid the event modification after every IO completion.